### ISSUE TITLE
Darknet model trained only with Pedestrian examples from COCO dataset performs very bad Pedestrian detection compared to pre-trained model weights from full 80 classes COCO dataset

### ISSUE THREAD

**Issue by MarlonCajamarca** [AlexeyAB/darknet#4532]
Hi all!
Recently, I used the [pre-trained yolo_v3.weights](https://pjreddie.com/media/files/yolov3.weights) trained using the 80 standart classes and then I post-process the detection results to contain detections from the PEDESTRIAN class, the class I want to detect, in order to evaluate the PEDESTRIAN detection accuracy of the given pre-trained model. 

The results were very good, therefore, I used the COCO API to create a custom COCO-PEDESTRIAN dataset, containing only the PEDESTRIAN examples from the full COCO dataset, and then I trained a custom yolo_v3 model using that COCO-PEDESTRIAN dataset. My initial plan was to replicate (or even surpass) the PEDESTRIAN detection accuracy of the standard pre-trained weights, however, the results were very poor using the COCO-PEDESTRIAN dataset. This is the mAP/loss curve from that training:
 
![image](https://user-images.githubusercontent.com/29712145/70954650-158e5f00-203c-11ea-86b8-8d8c829265f5.png)

I used the default network hyperparameters available in the [yolo_v3.cfg file](https://github.com/AlexeyAB/darknet/blob/master/cfg/yolov3.cfg) (except for the default anchors which were calculated again for the custom COCO-PEDESTRIAN dataset). I trained the model for 3 days in a Titan XP GPU until loss/mAP convergence.
Pre-trained COCO  example frame results:
![Captura de pantalla de 2019-12-16 20-12-53](https://user-images.githubusercontent.com/29712145/70955951-80419980-2040-11ea-8b54-0567435c568b.png)
Custom trained COCO-PEDESTRIAN same example detection results:
![Captura de pantalla de 2019-12-16 20-16-24](https://user-images.githubusercontent.com/29712145/70956113-fc3be180-2040-11ea-9cde-b7a6fcfd783a.png)

I expected a similar or better model detection performance once I train with only that specific class, compared to training on the full 80 classes. I am puzzled about why this happens? Why the pre-trained model, trained on several classes and with the same PEDESTRIAN examples, gave me better PEDESTRIAN detection results than the custom and targeted model trained on the specific COCO-PEDESTRIAN dataset? Are the new calculated anchors playing a role in that difference in the results pedestrian detection results? I would accept any help or advisory about how to cope with that issue. 

Thanks in advance!

**Comment by hwijune** [comment]
@MarlonCajamarca 

Did you check the anchor box?

**Comment by AlexeyAB** [comment]
* What command did you use for training?

* What params did you use in the Makefile? Did you compile Darknet with OpenCV?
* You should achieve at least `71.81% AP@50` for Person as in the default yolov3.weights/cfg. So just train at least 80 000 - 100 000 iterations
* Attach your cfg-file in zip

**Comment by MarlonCajamarca** [comment]
> @MarlonCajamarca
> 
> Did you check the anchor box?

@hwijune Yes, I calculated the anchor boxes again on the COCO-PEDESTRIAN dataset using the calc_anchors script provided. That is the only thing that differs from the [generic yolov3.cfg](https://github.com/AlexeyAB/darknet/blob/master/cfg/yolov3.cfg) file.

**Comment by MarlonCajamarca** [comment]
> * What command did you use for training?
> * What params did you use in the Makefile? Did you compile Darknet with OpenCV?
> * You should achieve at least `71.81% AP@50` for Person as in the default yolov3.weights/cfg. So just train at least 80 000 - 100 000 iterations
> * Attach your cfg-file in zip

Hi @AlexeyAB!

Firstly, thanks for the fast response! I detected the training input image resolution width and height parameters were set as width=512, height=512. Therefore, I stopped the training and started a new one with only this modification. In this moment, this is the Loss/mAP plot from current training (it seems is working well and reaching slowly the 71~72@0.5 mAP) and I am following your advice about training for minimum 80000 iterations to get the desired results. This is the plot:
![image](https://user-images.githubusercontent.com/29712145/71278391-77ddad00-2325-11ea-8c9b-bf8b16d0978e.png)
I will wait until new training finishes to do some tests. I will report back the results of the training in this thread.
Thanks!

**Comment by AlexeyAB** [comment]
@MarlonCajamarca 

Just you didn't set  `steps=80000,90000` (80% and 90% of max_batches): https://github.com/AlexeyAB/darknet#how-to-train-to-detect-your-custom-objects
This would increase accuracy at the end of training.

So you can stop training, set it and continue training with the last-trained file from /backup/

**Comment by AleBasso80** [comment]
Hello,
would it be possible just to use the pretrained weights and to limit the search to the person class without any further fine tuning or training?
